api/v1alpha1/loadbalancer_types.go

```diff
@@ -52,9 +52,27 @@ const (
 )
 
 // ConsistentHash defines the configuration related to the consistent hash
-// load balancer policy
+// load balancer policy.
+// +union
+//
+// +kubebuilder:validation:XValidation:rule="self.type == 'Header' ? has(self.header) : !has(self.header)",message="If consistent hash type is header, the header field must to be set."
 type ConsistentHash struct {
```

Comment:
I've turned this into a union to configure the different hash policies.

I envision future policies will added as additional union variants:

```go
type ConsistentHash struct {
  Type ConsistentHashType `json:"type"`
  Header *Header          `json:"header"`
  Cookie *Cookie          `json:"cookie"`
  // And so on
}
```